lib/resty/healthcheck.lua

```diff
@@ -315,15 +399,11 @@ end
 -- @return The results of the function; or nil and an error message
 -- in case it fails locking.
 local function locking_target_list(self, fn)
+  local ok, err = run_locked(self, self.TARGET_LIST_LOCK, with_target_list, self, fn)
 
-  local ok, err = run_fn_locked_target_list(false, self, fn)
-  if err == "failed to acquire lock" then
-    local _, terr = ngx_timer_at(0, run_fn_locked_target_list, self, fn)
-    if terr ~= nil then
-      return nil, terr
-    end
-
-    return true
+  if not ok and err == "scheduled" then
+    self:log(DEBUG, "target_list function re-scheduled in timer")
+    ok = true
   end
```

Comment:
There's no need to reset `ok` to anything if we use it to encode that the function was successfully scheduled for later.

```suggestion
  if ok == "scheduled" then
    self:log(DEBUG, "target_list function re-scheduled in timer")
  end
```